### PR TITLE
refactor(components, app): odd and desktop modal unification

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-619

# Overview

The `LegacyModal` , shell and header are going to be used for the PD liquid modal and the PD step list modal so a handful of changes were made:

1. `LegacyModal` is now `Modal` and lives in the component directory along with `LegacyModalShell`, `LegacyModalHeader` - both of which also lost `Legacy` at the beginning
2. `Modal` in the app directory is now `OddModal` and `ModalHeader` is now `OddModalHeader`
3. `Modal` in the components directory is now `LegacyModal` and has a deprecated tag.
4. some unused modals have been deleted which include: `interstitial`, `modalShell` in the components directory, `BaseModal`

## Changelog

basically what is outlined above! fixed all associated components

## Review requests

do a smoke test in the app and make sure it works as expected.

## Risk assessment

low

### CODE REVIEW

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Okay, but if we're going to use it for new stuff can we please call it something other than `LegacyModal`? I know we have a `Modal` in the app molecules already, so maybe we can figure out some other way to differentiate? What is the difference, anyway? If `app/src/molecules/Modal` is ODD-only, can we rename it to `ODDModal` perhaps?

**Comment by jerader** [comment]
> Okay, but if we're going to use it for new stuff can we please call it something other than `LegacyModal`? I know we have a `Modal` in the app molecules already, so maybe we can figure out some other way to differentiate? What is the difference, anyway? If `app/src/molecules/Modal` is ODD-only, can we rename it to `ODDModal` perhaps?

@sfoster1 both are good questions. Was blindly moving `LegacyModal` over so PD can use it in the future and did not think about renaming or other components. I'll post in the helix channel!

File: app/src/molecules/OddModal/__tests__/Modal.test.tsx
Comment (mjhuff) [inline app/src/molecules/OddModal/__tests__/Modal.test.tsx:1]: Does this test file need to be `OddModal.test.tsx`?

File: app/src/molecules/OddModal/__tests__/ModalHeader.test.tsx
Comment (mjhuff) [inline app/src/molecules/OddModal/__tests__/ModalHeader.test.tsx:1]: Should this be `OddModalHeader.test.tsx`?

File: app/src/molecules/OddModal/__tests__/Modal.test.tsx
Comment (mjhuff) [inline app/src/molecules/OddModal/__tests__/Modal.test.tsx:1]: Should this file be `OddModal.test.tsx`?

**Review by mjhuff — COMMENTED** [review]
Just a few comments! Will look more as I have time.

Any chance we can leave a docstring on the Modal that we are using for the desktop that it is meant only for the desktop?

File: app/src/molecules/OddModal/__tests__/ModalHeader.test.tsx
Comment (jerader) [reply app/src/molecules/OddModal/__tests__/ModalHeader.test.tsx:1]: oops yes you are right, thanks!

**Comment by jerader** [comment]
> Any chance we can leave a docstring on the Modal that we are using for the desktop that it is meant only for the desktop?

@mjhuff yes, i left a comment on the `Modal` component that says 

```
/**
 * For Desktop app and Helix (which includes Protocol Designer) use only.
 */
```

File: components/src/modals/index.tsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-export interface LegacyModalProps extends StyleProps {
+// modal components
+export * from './AlertModal'
+export * from './ContinueModal'
+export * from './LegacyModal'
+export * from './Overlay'
+export * from './SpinnerModal'
+export * from './SpinnerModalPage'
+export * from './ModalShell'
+export * from './ModalHeader'
```
Comment (koji) [inline components/src/modals/index.tsx]: maybe creating `index.ts` for export statements?

File: app/src/organisms/DeviceDetailsDeckConfiguration/AddFixtureModal.tsx [hunk 2]
```diff
@@ -49,17 +50,16 @@ import {
...
 
 import type {
   CutoutConfig,
   CutoutId,
   CutoutFixtureId,
 } from '@opentrons/shared-data'
-import type { ModalHeaderBaseProps } from '../../molecules/Modal/types'
-import type { LegacyModalProps } from '../../molecules/LegacyModal'
+import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
+import type { ModalProps } from '@opentrons/components'
```
Comment (koji) [inline app/src/organisms/DeviceDetailsDeckConfiguration/AddFixtureModal.tsx]:
```suggestion
import type { ModalProps } from '@opentrons/components'
import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
```

File: app/src/organisms/DeviceDetailsDeckConfiguration/DeckFixtureSetupInstructionsModal.tsx [hunk 3]
```diff
@@ -8,14 +8,14 @@ import {
...
 } from '@opentrons/components'
 import { ExternalLink } from '../../atoms/Link/ExternalLink'
-import { Modal } from '../../molecules/Modal'
-import { LegacyModal } from '../../molecules/LegacyModal'
+import { OddModal } from '../../molecules/OddModal'
 
-import type { ModalHeaderBaseProps } from '../../molecules/Modal/types'
-import type { LegacyModalProps } from '../../molecules/LegacyModal'
+import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
+import type { ModalProps } from '@opentrons/components'
```
Comment (koji) [inline app/src/organisms/DeviceDetailsDeckConfiguration/DeckFixtureSetupInstructionsModal.tsx]:
```suggestion
import type { ModalProps } from '@opentrons/components'
import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
```

File: app/src/organisms/EmergencyStop/EstopMissingModal.tsx [hunk 4]
```diff
@@ -10,15 +10,15 @@ import {
...
 import { getTopPortalEl } from '../../App/portal'
-import { LegacyModal } from '../../molecules/LegacyModal'
-import { Modal } from '../../molecules/Modal'
+import { OddModal } from '../../molecules/OddModal'
 import { getIsOnDevice } from '../../redux/config'
 
-import type { ModalHeaderBaseProps } from '../../molecules/Modal/types'
-import type { LegacyModalProps } from '../../molecules/LegacyModal'
+import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
+import type { ModalProps } from '@opentrons/components'
```
Comment (koji) [inline app/src/organisms/EmergencyStop/EstopMissingModal.tsx]:
```suggestion
import type { ModalProps } from '@opentrons/components'
import type { OddModalHeaderBaseProps } from '../../molecules/OddModal/types'
```

File: app/src/organisms/EmergencyStop/EstopPressedModal.tsx [hunk 5]
```diff
@@ -25,15 +26,14 @@ import { getTopPortalEl } from '../../App/portal'
...
 import { getIsOnDevice } from '../../redux/config'
 
 import type {
-  ModalHeaderBaseProps,
+  OddModalHeaderBaseProps,
   ModalSize,
-} from '../../molecules/Modal/types'
-import type { LegacyModalProps } from '../../molecules/LegacyModal'
+} from '../../molecules/OddModal/types'
+import type { ModalProps } from '@opentrons/components'
```
Comment (koji) [inline app/src/organisms/EmergencyStop/EstopPressedModal.tsx:36]:
```ts
import type { ModalProps } from '@opentrons/components'
import type {
  OddModalHeaderBaseProps,
  ModalSize,
} from '../../molecules/OddModal/types'
```

File: app/src/organisms/OnDeviceDisplay/RunningProtocol/RunFailedModal.tsx [hunk 6]
```diff
@@ -17,9 +17,9 @@ import {
 import { useStopRunMutation } from '@opentrons/react-api-client'
 
 import { SmallButton } from '../../../atoms/buttons'
-import { Modal } from '../../../molecules/Modal'
+import { OddModal } from '../../../molecules/OddModal'
 
-import type { ModalHeaderBaseProps } from '../../../molecules/Modal/types'
+import type { OddModalHeaderBaseProps } from '../../../molecules/OddModal/types'
 import type { RunError } from '@opentrons/api-client'
```
Comment (koji) [inline app/src/organisms/OnDeviceDisplay/RunningProtocol/RunFailedModal.tsx]:
```suggestion
import type { OddModalHeaderBaseProps } from '../../../molecules/OddModal/types'
import type { RunError } from '@opentrons/api-client'
```
```suggestion
import type { RunError } from '@opentrons/api-client'
import type { OddModalHeaderBaseProps } from '../../../molecules/OddModal/types'
```

**Review by koji — APPROVED** [review]
Left some comments.
Overall, the changes look good to me. Thank you for doing this!

File: components/src/modals/index.tsx (hunk 1 above)
Comment (jerader) [reply components/src/modals/index.tsx]: good call, i'll move the modal component in there out into its own file and change this to index.ts

**Review by sfoster1 — APPROVED** [review]
Awesome! I am unreasonably happy about this. Thank you so much!

**Review by mjhuff — APPROVED** [review]
Thank you for doing this! Looks great!
